ui/app/pages/permissions-connect/permissions-connect.component.js

```diff
@@ -53,43 +49,75 @@ export default class PermissionConnect extends Component {
     selectedAccountAddresses: this.props.accounts.length === 1
       ? new Set([this.props.accounts[0].address])
       : new Set(),
-    permissionAccepted: null,
+    permissionsApproved: null,
     origin: this.props.origin,
+    targetDomainMetadata: this.props.targetDomainMetadata || {},
   }
 
   beforeUnload = () => {
     const { permissionsRequestId, rejectPermissionsRequest } = this.props
-    const { permissionAccepted } = this.state
+    const { permissionsApproved } = this.state
 
-    if (permissionAccepted === null && permissionsRequestId) {
+    if (permissionsApproved === null && permissionsRequestId) {
       rejectPermissionsRequest(permissionsRequestId)
     }
   }
 
   removeBeforeUnload = () => {
     const environmentType = getEnvironmentType()
-    if (
-      environmentType === ENVIRONMENT_TYPE_FULLSCREEN ||
-      environmentType === ENVIRONMENT_TYPE_NOTIFICATION
-    ) {
+    if (environmentType === ENVIRONMENT_TYPE_NOTIFICATION) {
       window.removeEventListener('beforeunload', this.beforeUnload)
     }
   }
 
+  componentDidMount () {
+    const {
+      getCurrentWindowTab,
+      getRequestAccountTabIds,
+      permissionsRequest,
+      history,
+    } = this.props
+    getCurrentWindowTab()
+    getRequestAccountTabIds()
+
+    if (!permissionsRequest) {
+      return history.push(DEFAULT_ROUTE)
+    }
+
+    const environmentType = getEnvironmentType()
+    if (environmentType === ENVIRONMENT_TYPE_NOTIFICATION) {
+      window.addEventListener('beforeunload', this.beforeUnload)
+    }
```

Comment:
`componentDidMount` was moved. The only functional change is here: only add the event listener for the notification UI.